File: 2015/day02-2.go
package main

import (
	"fmt"
	"os"
	"slices"
	"strconv"
	"strings"
)

func main() {
	content, _ := os.ReadFile("./data/input2.txt")
	lines := strings.Split(string(content), "\n")
	total := 0
	for _, line := range lines[0 : len(lines)-1] {
		split := strings.Split(line, "x")
		dimensions := make([]int, len(split))
		for i, item := range split {
			dimensions[i], _ = strconv.Atoi(item)
		}
		slices.Sort(dimensions)

		ribbon_total := (2 * dimensions[0]) + (2 * dimensions[1]) + (dimensions[0] * dimensions[1] * dimensions[2])
		total += ribbon_total
	}

	fmt.Println(total)
}

File: 2017/day02-2.js
import fs from 'fs'

const data = fs.readFileSync('./data/input2.txt', 'utf-8')

const lines = data.split('\n')
lines.pop()

let checksum = 0
for (const line of lines) {
  const nums = line.split('\t').map(x => parseInt(x))
  nums.sort((a, b) => a - b)
  let found = false
  let i = nums.length - 1
  while (!found) {
    const num = nums[i]
    for (let j = i - 1; j >= 0; j--) {
      const compare = nums[j]
      const remainder = num % compare
      if (remainder === 0) {
        checksum += (num / compare)
        found = true
      }
    }
    i--
  }
}

console.log(checksum)

